src/main/java/net/minecraftforge/server/permission/handler/IPermissionHandler.java

```diff
@@ -0,0 +1,73 @@
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.server.permission.handler;
+
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.util.StringRepresentable;
+import net.minecraftforge.server.permission.PermissionAPI;
+import net.minecraftforge.server.permission.events.PermissionGatherEvent;
+import net.minecraftforge.server.permission.nodes.PermissionDynamicContext;
+import net.minecraftforge.server.permission.nodes.PermissionNode;
+
+import java.util.Collection;
+
+/**
+ * This is the Heart of the PermissionAPI, it manages {@link PermissionNode}s
+ * as well as it handles all permission queries.
+ *
+ * <p>Note: You do not need to implement a PermissionHandler to query for permissions.</p>
+ *
+ * @implNote The {@link DefaultPermissionHandler} does forward all permission queries to the PermissionNodes default resolver.
+ * @apiNote You can implement your own PermissionHandler using the {@link PermissionGatherEvent.Handler} event.
+ */
+public interface IPermissionHandler
+{
+   /**
+    * {@return an identifier for the PermissionHandler, used for logging purposes}
+    */
+   ResourceLocation getPermissionHandlerName();
+
+   /**
+    * @param node to be registered, should not be called by mods.
+    */
+   void registerNode(PermissionNode<?> node);
+
+   /**
+    * {@return a collection of all registered Nodes}
+    */
+   Collection<PermissionNode<?>> getRegisteredNodes();
+
+   /**
+    * <p><strong>Use {@link PermissionAPI#getPermissions(ServerPlayer, PermissionNode, PermissionDynamicContext[])}</strong></p>
+    *
+    * <p>Queries a player's permission for a given node and contexts</p>
+    * <p><strong>Warning:</strong> PermissionNodes <strong>must</strong> be registered using the
+    * {@link PermissionGatherEvent.Nodes} event before querying.</p>
+    *
+    * @param player player for which you want to check permissions
+    * @param node the PermissionNode for which you want to query
+    * @param context optional array of PermissionDynamicContext, single entries will be ignored if they weren't
+    *                registered to the node
+    * @param <T> type of the queried PermissionNode
+    * @return a value of type {@code <T>}, that the combination of Player and PermissionNode map to.
+    */
+   <T> T getPermissions(ServerPlayer player, PermissionNode<T> node, PermissionDynamicContext<? extends StringRepresentable>... context);
```

Comment:
I'd suggest losing the `s`: `getPermission`, since there's only one permission node and corresponding value here.